baseplate/diagnostics/tracing/publisher.py

```diff
@@ -0,0 +1,191 @@
+                response = getattr(exc, "response", None)
+                if response is not None:
+                    logger.exception("HTTP Request failed. Error: %s", response.text)
+                    # If client error, crash
+                    if response.status_code < 500:
+                        raise
+                else:
+                    logger.exception("HTTP Request failed. Response not available")
+            except IOError:
+                self.metrics.counter("error.io").increment()
+                logger.exception("HTTP Request failed")
+            else:
+                self.metrics.counter("sent").increment(payload.count)
+                return
+
+            tries += 1
+            if tries > self.retry_limit:
+                logger.error("Retry limit hit")
+                raise RetryLimitExceeded()
+            else:
+                logger.info("Retrying batch send. Try number %s", tries)
+                time.sleep(self.retry_delay * tries)
+
+
+def publish_traces():
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument("config_file", type=argparse.FileType("r"),
+        help="path to a configuration file")
+    arg_parser.add_argument("--queue-name", default="main",
+        help="name of trace queue / publisher config (default: main)")
+    arg_parser.add_argument("--debug", default=False, action="store_true",
+        help="enable debug logging")
+    arg_parser.add_argument("--app-name", default="main", metavar="NAME",
+        help="name of app to load from config_file (default: main)")
+    args = arg_parser.parse_args()
+
+    if args.debug:
+        level = logging.DEBUG
+    else:
+        level = logging.WARNING
+    logging.basicConfig(level=level)
+
+    config_parser = configparser.RawConfigParser()
+    config_parser.readfp(args.config_file)
+
+    app_raw_config = dict(config_parser.items("app:{}".format(args.app_name)))
+    app_cfg = config.parse_config(app_raw_config, {
+        "tracing": {
+            "endpoint": config.Optional(config.Endpoint),
+            "queue_name": config.Optional(config.String)
+        }
+    })
+
+    # pylint: disable=maybe-no-member
+    assert args.queue_name == app_cfg.tracing.queue_name, \
+        "Queue name command line arg ({}) doesn't match tracing configuration ({})".format(
+            args.gueue_name, app_cfg.queue_name)
+
+    publisher_raw_cfg = dict(config_parser.items("tracing-publisher:" + args.queue_name))
+    publisher_cfg = config.parse_config(publisher_raw_cfg, {
+        "post_timeout": config.Optional(config.Integer, POST_TIMEOUT_DEFAULT),
+        "max_batch_size": config.Optional(config.Integer, MAX_BATCH_SIZE_DEFAULT),
+        "retry_delay": config.Optional(config.Integer, RETRY_DELAY_DEFAULT),
+        "retry_limit": config.Optional(config.Integer, RETRY_LIMIT_DEFAULT),
+        "zipkin_api_version": config.Optional(config.String, ZIPKIN_API_VERSION_DEFAULT),
```

Comment:
Is there any advantage to breaking the URL components all out in the config instead of just asking for a full URL? I imagine if the API version were different we'd need to change code here anyway?

Reply:
> s there any advantage to breaking the URL components all out in the config instead of just asking for a full URL? 

I was trying to maintain some backwards compatibility with the in-app approach which only asks for an hostname and optional port.

> I imagine if the API version were different we'd need to change code here anyway?

The POST to the ``/spans`` endpoint is the same in both V1 and V2 so in theory, a config change is all that would be needed to use V2.

I think a move to a single ``zipkin_api_url`` (example value -- ``https://zipkin.snooguts.net:9411/api/v2``) is best. However, I was hesitant to introduce two different configuration paths while the in-app trace publishing is still supported.

Reply:
Totally appreciated on trying to stay consistent. In this case, it's gonna be in a separate section of config even so it seems like a reasonable place to break backwards-compatibility/consistency if we're agreed that full URL is more future-flexible / easier to deal with.